File: src/main/java/eu/happybe/openapi/stats/Stats.java
package eu.happybe.openapi.stats;

import cn.nukkit.Player;
import eu.happybe.openapi.mysql.QueryQueue;
import eu.happybe.openapi.mysql.query.AddExperienceQuery;

public class Stats {

    public static void addExperience(Player player, int experience) {
        QueryQueue.submitQuery(new AddExperienceQuery(player.getName(), experience), query -> {
            if(!(query instanceof AddExperienceQuery)) {
                return;
            }

            if(((AddExperienceQuery) query).levelUp && player.isOnline()) {
                player.sendMessage("§9HappyBedrock> §a§lLEVEL UP! §r§aCurrent level: {$query->newLevel}!");
                player.namedTag.putInt("HappyBedrockLevel", ((AddExperienceQuery) query).newLevel);
            }
        });
    }
}

File: src/main/java/eu/happybe/openapi/stats/Stats.kt
package eu.happybe.openapi.stats

import cn.nukkit.Player
import eu.happybe.openapi.mysql.QueryQueue
import eu.happybe.openapi.mysql.query.AddPointQuery

enum class Stats(@field:Getter private val columnName: String, @field:Getter private val table: String) {
    POINT_CRYSTAL("Crystals", "Values"), POINT_UHCRUN_WIN("UHCRunWins", "Stats"), POINT_UHCRUN_KILL("UHCRunKills", "Stats");

    companion object {
        fun addPoint(player: Player, stat: Stats) {
            QueryQueue.submitQuery(AddPointQuery(player.name, stat.getColumnName(), stat.getTable()))
        }
    }
}